### PR TITLE
Add 'Cleanup Snapshot' API Documentation

### PR DESCRIPTION
**Pull request by Reiyadowns**
### Description
Adds documentation for the 'Cleanup Snapshot' API

### Issues Resolved
Addresses half of issue #2111
(In combination with the branch 'add-clone-snapshot-docs', the issue is fully addressed and can be closed)

### Version
Applies to all versions.

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by Naarcha-AWS** [comment]
@Reiyadowns: Thank you for your contribution! I've added a few additional tweaks on top and will not pass this along to the our editor.

**Review by Naarcha-AWS — APPROVED** [review]

File: _api-reference/snapshots/cleanup-snapshot-repository.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+		"deleted_blobs":8
+	}
+}
+```
+
+## Response body fields
+
+| Field | Data type | Description |
+| :--- | :--- | :--- |
+| `deleted_bytes` | Integer | The amount of bytes freed after data deletion. |
```
Comment (natebower) [inline _api-reference/snapshots/cleanup-snapshot-repository.md:62]:
```suggestion
| `deleted_bytes` | Integer | The number of bytes freed after data deletion. |
```

File: _api-reference/snapshots/cleanup-snapshot-repository.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+		"deleted_bytes":40,
+		"deleted_blobs":8
+	}
+}
+```
+
+## Response body fields
+
+| Field | Data type | Description |
+| :--- | :--- | :--- |
```
Comment (natebower) [inline _api-reference/snapshots/cleanup-snapshot-repository.md:61]: Below: Can we use something more precise than "freed"? Do we mean something like "The number of bytes made available by data deletion"?

File: _api-reference/snapshots/cleanup-snapshot-repository.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+| `deleted_bytes` | Integer | The amount of bytes freed after data deletion. |
+| `deleted_blobs` | Integer | The amount of binary large objects (BLOBs) cleared from the repository by the request. |
```
Comment (natebower) [inline _api-reference/snapshots/cleanup-snapshot-repository.md:63]:
```suggestion
| `deleted_blobs` | Integer | The number of binary large objects (BLOBs) cleared from the repository by the request. |
```

**Review by natebower — COMMENTED** [review]
@Reiyadowns @Naarcha-AWS Please see my comment/changes and let me know if you have any questions. Thanks!

File: _api-reference/snapshots/cleanup-snapshot-repository.md (hunk 1 above)
Comment (Naarcha-AWS) [inline _api-reference/snapshots/cleanup-snapshot-repository.md:62]: @natebower: Does this work?

```suggestion
| `deleted_bytes` | Integer | The number of bytes now available in the snapshot after data deletion. |
```

File: _api-reference/snapshots/cleanup-snapshot-repository.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+		"deleted_blobs":8
+	}
+}
+```
+
+## Response body fields
+
+| Field | Data type | Description |
+| :--- | :--- | :--- |
+| `deleted_bytes` | Integer | The number of bytes now available in the snapshot after data deletion. |
```
Comment (Naarcha-AWS) [inline _api-reference/snapshots/cleanup-snapshot-repository.md:62]:
```suggestion
| `deleted_bytes` | Integer | The number of bytes made available in the snapshot after data deletion. |
```

**Pull request merged**